core/src/main/java/com/datastrato/gravitino/listener/api/event/AlterMetalakeEvent.java

```diff
@@ -0,0 +1,58 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.listener.api.event;
+
+import com.datastrato.gravitino.MetalakeChange;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.annotation.DeveloperApi;
+import com.datastrato.gravitino.listener.api.info.MetalakeInfo;
+
+/** Represents an event fired when a metalake is successfully altered. */
+@DeveloperApi
+public final class AlterMetalakeEvent extends MetalakeEvent {
+  private final MetalakeInfo updatedMetalakeInfo;
+  private final MetalakeChange[] metalakeChanges;
+
+  /**
+   * Constructs an instance of {@code AlterMetalakeEvent}, encapsulating the key details about the
+   * successful alteration of a metalake.
+   *
+   * @param user The username of the individual responsible for initiating the metalake alteration.
+   * @param identifier The unique identifier of the altered metalake, serving as a clear reference
+   *     point for the metalake in question.
+   * @param metalakeChanges An array of {@link MetalakeChange} objects representing the specific
+   *     changes applied to the metalake during the alteration process.
+   * @param updatedMetalakeInfo The post-alteration state of the metalake.
+   */
+  public AlterMetalakeEvent(
+      String user,
+      NameIdentifier identifier,
+      MetalakeChange[] metalakeChanges,
+      MetalakeInfo updatedMetalakeInfo) {
+    super(user, identifier);
+    this.metalakeChanges = metalakeChanges.clone();
```

Comment:
This method `clone` implementation is indeterminate and I think it will be problematic.

Reply:
We should do a array copy for an array, right?

Reply:
yes,  `clone` shadow copies the array like `System.arraycopy`, I think this's enough

Reply:
@yuqi1129 what do you think?

Reply:
I don't think current XXXChange API has the ability to modify the data.